includes/Core/Authentication/Clients/Client_Factory.php

```diff
@@ -149,6 +149,56 @@ private static function get_http_client_config( $config ) {
 			$config['proxy'] = "{$auth}{$http_proxy->host()}:{$http_proxy->port()}";
 		}
 
+		// Respect WordPress HTTP request blocking settings.
+		if ( defined( 'WP_HTTP_BLOCK_EXTERNAL' ) && WP_HTTP_BLOCK_EXTERNAL ) {
+			// Get the site host.
+			$site_host = wp_parse_url( home_url(), PHP_URL_HOST );
+
+			// Get allowed hosts from WP_ACCESSIBLE_HOSTS.
+			$allowed_hosts = array( 'localhost', '127.0.0.1', $site_host );
+			if ( defined( 'WP_ACCESSIBLE_HOSTS' ) && WP_ACCESSIBLE_HOSTS ) {
+				$user_allowed_hosts = explode( ',', WP_ACCESSIBLE_HOSTS );
+				$allowed_hosts      = array_merge( $allowed_hosts, array_map( 'trim', $user_allowed_hosts ) );
+			}
+
+			// Create a handler stack with middleware to check hosts.
+			$stack = \Google\Site_Kit_Dependencies\GuzzleHttp\HandlerStack::create();
```

Comment:
Please use import.

Also, should we should probably be pushing the added middleware onto the current/default stack rather than replacing it. If we're not blocking the request, we'd want it to flow through the stack as usual.